types/underscore/underscore-tests.ts

```diff
@@ -571,6 +571,31 @@ function chain_tests() {
 		.map(num => "string" + num)
 		.value();
 
+  var numEvens = _.chain(['a', 'bb', 'ccc', 'dddd'])
+    .map(n => n.length)
+    .filter(s => s % 2 == 0)
+    .value()
+    .length;
+
+  var sortBy = _.chain([4, 3, 2, 1])
+    .map(v => v * 5)
+    .sortBy(v => v)
+    .value()
+    .length;
+
+  var each = _.chain([4, 3, 2, 1])
+    .map(v => v * 5)
+    .each(v => console.log(v))
+    .value()
+    .length;
+
+  var groupBy = _.chain(['a', 'aa', 'b', 'bb'])
+    .groupBy(v => v[0])
+    .value()[0].length;
+
+  var sampleOne = _.chain([1, 2, 3]).sample().value() * 2;
+  var sampleMore = _.chain([1, 2, 3]).sample(2).value().length;
+
```

Comment:
These tests are good, but they don't cover all the methods you have fixed. Please try to come up with a test for each method that you have edited, preferably one that would have failed before your edits.

Reply:
I agree!